### PR TITLE
Add mushroom trees to fungus grower

### PR DESCRIPTION
**Pull request by Diet-Cola**
https://github.com/CivMC/Civ/issues/200

### CODE REVIEW

**Review by psygate — APPROVED** [review]
I guess two lines is good. I'm not to deep into RB but if this is all that it takes. Take my approval and go with god.

**Comment by Aleksey-Terzi** [comment]
**block.setType(Material.AIR);** needs to be added otherwise sir refuses to grow.
```
		final ServerLevel world = ((CraftWorld) block.getWorld()).getHandle();
		final BlockPos position = new BlockPos(block.getX(), block.getY(), block.getZ());
		//Taken from CraftWorld.generateTree()
		block.setType(Material.AIR);
		if (!growth.value().place(world, world.getChunkSource().getGenerator(), this.random, position)) {
			block.setType(material);
		}
```

Also, I suggest replacing BROWN_MUSHROOM in the default config by example for a huge brown mushroom:
```
  BROWN_MUSHROOM:
    persistent_growth_period: 12h
    name: Huge Brown Mushroom
    id: 24
    grower:
      type: fungus
      material: BROWN_MUSHROOM
    vanilla_materials:
      - BROWN_MUSHROOM
    item:
      ==: org.bukkit.inventory.ItemStack
      type: BROWN_MUSHROOM
      v: 2730
    needs_sun_light: false
    biomes:
      swampy: 2.0
      foresty: 1.0
      cold_forest: 1.0
      freshwater: 1.0
```

File: paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java [hunk 1]
```diff
@@ -55,6 +55,7 @@ public void setStage(final Plant plant, final int stage) {
 		final ServerLevel world = ((CraftWorld) block.getWorld()).getHandle();
 		final BlockPos position = new BlockPos(block.getX(), block.getY(), block.getZ());
 		//Taken from CraftWorld.generateTree()
+		block.setType(Material.AIR);
```
Comment (psygate) [inline paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java:58]: Are we sure you can't remove reinforcements with this?

Comment (Aleksey-Terzi) [reply paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java:58]: Plants cannot be reinforced

File: paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java [hunk 2]
```diff
@@ -46,13 +46,16 @@ public void setStage(final Plant plant, final int stage) {
 				material == Material.CRIMSON_FUNGUS ? TreeFeatures.CRIMSON_FUNGUS :
```
Comment (psygate) [inline paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java:46]: Can we please at the same time remove the ternary operator here, and make it something more sensible? This is completely unreadable and opaque code.

**Review by psygate — CHANGES_REQUESTED** [review]

File: paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java (hunk 2 above)
Comment (Aleksey-Terzi) [reply paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java:46]: I suggest separating refactoring of unrelated (to fix) code and refactoring.

Comment (Aleksey-Terzi) [reply paper/src/main/java/com/untamedears/realisticbiomes/growth/FungusGrower.java:46]: I mean can we make refactoring as a separate PR.

**Pull request closed without merging**